### PR TITLE
Add feature extraction document for luigi

### PR DESCRIPTION
**Pull request by umursen**

### CODE REVIEW

File: docs/general/sentence_processing_and_feature_extraction_techniques_for_luigi.md [hunk 1]
```diff
@@ -0,0 +1,109 @@
...
+| Triple Extraction                 | `nlp.prop_triples`            | A triple consists of subject, predicate, object of sentence | Triple: subject: 'Revolutions', predicate: 'need', object: 'chickens' |
+| About Roboy                       | `nlp.prop_roboy`              | Detecting whether sentence is about Roboy                   | 'you', 'roboy', 'robot', 'roboboy', ... |
+| Yes-No                            | `nlp.prop_yesno`              | Detecting answers to yes-no questions                       | Checking for 'yes', 'no', 'i don't know', 'probably', 'probably not' and synonyms of these                           |
+| Sentence Type: Question           | `nlp.sig_is_question`         | Emitted if the input sentence is a question                 |                                                |
+| Play Game                         | `nlp.sig_intent_play`         | Emitted when the interlocutor wants to play a game          | input: "I want to play", "I like games" or something similar    |
+
+### Using the Features
+
+#### Detailed POS Tagging
```
Comment (josephbirkner) [inline docs/general/sentence_processing_and_feature_extraction_techniques_for_luigi.md]: This is not entirely correct - Triples are already build on top of this. For example, if you want to match an ice cream flavor, you can use `ctx[nlp.prop_triples][0].match_either_lemma(obj={"vanilla", "strawberry", "lemon", ...})`.

If you want to detect questions, this is also built into triple - use `ctx[nlp.prop_triples][0].is_question()`, and the get the question word either from `ctx[nlp.prop_triples][0].get_subject()` or `ctx[nlp.prop_triples][0].get_object()`.

File: docs/general/sentence_processing_and_feature_extraction_techniques_for_luigi.md [hunk 2]
```diff
@@ -0,0 +1,109 @@
...
+or 
+
+- credit card
+- cash
+- bitcoin
+
+Alternatively by using POS, Roboy can understand whether the sentence is a request or not by only parsing the verb.
+For example if the customer uses one of the following phrases, it is easy to detect that customer has a request:
+
+- would like
```
Comment (josephbirkner) [inline docs/general/sentence_processing_and_feature_extraction_techniques_for_luigi.md]: Same here, use `match_either_lemma`.

File: docs/general/sentence_processing_and_feature_extraction_techniques_for_luigi.md [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+Lemmatization allows to get the base form of words which is the required form for matching.
+
+
+#### Named Entity Recognition (NER)
+
+NER can be used in order to understand the quantity of ice cream, payment or any matter that requires specifics.
+The following figure includes the most possible types of use, please check 
+https://spacy.io/api/annotation#named-entities for further types.
+
+| Type                              | DESCRIPTION
```
Comment (josephbirkner) [inline docs/general/sentence_processing_and_feature_extraction_techniques_for_luigi.md]: Please insert a reference to spacy's interactive NER detector here: https://explosion.ai/demos/displacy-ent

You will notice that it is not very reliable...

File: docs/general/sentence_processing_and_feature_extraction_techniques_for_luigi.md [hunk 4]
```diff
@@ -0,0 +1,109 @@
...
+| TIME                              | Times smaller than a day.
+| PERCENT                           | Percentage, including ”%“.                
+| MONEY                             | Monetary values, including unit.    
+| QUANTITY                          | Measurements, as of weight or distance.            
+| ORDINAL                           | “first”, “second”, etc.           
+| CARDINAL                          | Numerals that do not fall under another type.             
+
+
+### Triple Extraction
+Triple Extraction can be used to understand any customer request. In addition to Detailed POS,
```
Comment (josephbirkner) [inline docs/general/sentence_processing_and_feature_extraction_techniques_for_luigi.md]: This is a misleading sentence - Triples are just an abstraction on top of POS tags. Usually you don't try to parse the POS tags manually, but use the pre-processed form provided by triples.

File: docs/general/sentence_processing_and_feature_extraction_techniques_for_luigi.md [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+
+Example: State that reads the "yesno" property
+
+```python
+import ravestate as rs
+import ravestate_nlp as nlp
+import ravestate_rawio as rawio
+
+@rs.state(
+    cond=nlp.prop_yesno.changed(),  # state reacts to a change in the 'yesno' property
```
Comment (josephbirkner) [inline docs/general/sentence_processing_and_feature_extraction_techniques_for_luigi.md]: Maybe add a state that emit's some `question-asked` signal, to increase the value of this example: This state reacts when the user says yes or no, in response to some question asked before.

File: docs/general/sentence_processing_and_feature_extraction_techniques_for_luigi.md (hunk 1 above)
Comment (umursen) [reply docs/general/sentence_processing_and_feature_extraction_techniques_for_luigi.md]: I also thought about this solution but Triples is not working as i expected, it only captures the last word of the corresponding subject. For example if the string is 'i want vanilla ice cream', it extracts '[<Triple object i:want:cream>]'. On the other hand we can understand whether there is a request or not. Therefore i was planning to get the customer's intention with Triples and flavors with POS Tagging.

**Pull request merged**